### PR TITLE
Issue/1423

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is for #1423 

## How Has This Been Tested?
Tested this with 2 possible scenarios:
- API call for all forms (Made sure that goals appear for the forms where goal is set)
- Api call for individual form

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/api/class-give-api.php [hunk 1]
```diff
@@ -1037,6 +1037,13 @@ private function get_form_data( $form_info ) {
 			$form['info']['tags'] = get_the_terms( $form_info, 'give_forms_tag' );
 		}
 
+		if ( give_is_setting_enabled( give_get_meta( $form_info->ID, '_give_goal_option', true ) ) ) {
+			$goal_amount = give_get_meta( $form_info->ID, '_give_set_goal', true );
+			$goal_percentage_completed = round( ( give_get_form_earnings_stats( $form_info->ID ) / $goal_amount ) * 100 );
```
Comment (ravinderk) [inline includes/api/class-give-api.php]: @mehul0810 This code will generate PHP notice.
```
PHP Warning:  Division by zero in /wp-content/plugins/Give/includes/api/class-give-api.php on line 1042
```

And also max percentage should be `100`
for ref: https://github.com/WordImpress/Give/blob/master/templates/shortcode-goal.php#L50

Comment (mehul0810) [inline includes/api/class-give-api.php]: @ravinderk I've updated the code, considering both the above mentioned scenarios

File: includes/api/class-give-api.php [hunk 2]
```diff
@@ -1037,9 +1037,12 @@ private function get_form_data( $form_info ) {
 			$form['info']['tags'] = get_the_terms( $form_info, 'give_forms_tag' );
 		}
 
-		if ( give_is_setting_enabled( give_get_meta( $form_info->ID, '_give_goal_option', true ) ) ) {
-			$goal_amount = give_get_meta( $form_info->ID, '_give_set_goal', true );
-			$goal_percentage_completed = round( ( give_get_form_earnings_stats( $form_info->ID ) / $goal_amount ) * 100 );
+		// Check whether any goal is to be achieved for the donation form
+		$goal_option = give_get_meta( $form_info->ID, '_give_goal_option', true );
+		$goal_amount = give_get_meta( $form_info->ID, '_give_set_goal', true );
+		if ( give_is_setting_enabled( $goal_option ) && isset( $goal_amount ) &&  $goal_amount > 0 ) {
```
Comment (ravinderk) [inline includes/api/class-give-api.php]: @mehul0810 change this to `if ( give_is_setting_enabled( $goal_option ) && $goal_amount  )`

Comment (mehul0810) [inline includes/api/class-give-api.php]: @ravinderk I've implemented this change and liked it so much 👍

**Comment by DevinWalker** [comment]
Closing in favor of https://github.com/WordImpress/Give/pull/1792 with my code updates.

**Pull request closed without merging**